x-pack/plugin/slm/src/main/java/org/elasticsearch/xpack/slm/SlmHealthIndicatorService.java

```diff
@@ -255,4 +249,148 @@ private static HealthIndicatorDetails createDetails(
+            Long missingStartTime = getMissingSnapshotPeriodStartTime(policy);
+            return "- ["
+                + policy.getId()
+                + "] has not had a snapshot for "
+                + (policy.getPolicy().getMissingSnapshotUnhealthyThreshold() != null
+                    ? policy.getPolicy().getMissingSnapshotUnhealthyThreshold().toHumanReadableString(2)
+                    : "")
+                + (missingStartTime != null ? ", since [" + FORMATTER.formatMillis(missingStartTime) + "]" : "");
+        }).collect(Collectors.joining("\n"));
+        String cause = (unhealthyPolicies.size() > 1
+            ? "Several automated snapshot policies are unhealthy:\n"
+            : "An automated snapshot policy is unhealthy:\n") + unhealthyPolicyCauses;
+
+        String unhealthyPolicyActions = unhealthyPolicies.stream()
+            .map(policy -> "- GET /_slm/policy/" + policy.getId() + "?human")
+            .collect(Collectors.joining("\n"));
+        String action = "Check the snapshot lifecycle "
+            + (unhealthyPolicies.size() > 1 ? "policies" : "policy")
+            + " for detailed failure info:\n"
+            + unhealthyPolicyActions;
+
+        return createIndicator(
+            YELLOW,
+            "Encountered [" + unhealthyPolicies.size() + "] unhealthy snapshot lifecycle management policies",
+            createDetails(verbose, unhealthyPolicies, slmMetadata, currentMode),
+            impacts,
+            verbose
+                ? List.of(
+                    new Diagnosis(
+                        checkTroubleshootingGuide(cause, action),
+                        List.of(
+                            new Diagnosis.Resource(
+                                Diagnosis.Resource.Type.SLM_POLICY,
+                                unhealthyPolicies.stream()
+                                    .map(SnapshotLifecyclePolicyMetadata::getId)
+                                    .limit(Math.min(unhealthyPolicies.size(), maxAffectedResourcesCount))
+                                    .toList()
+                            )
+                        )
+                    )
+                )
+                : List.of()
+        );
+    }
+
+    private static Long getMissingSnapshotPeriodStartTime(SnapshotLifecyclePolicyMetadata policy) {
+        if (policy.getLastSuccess() != null) {
+            // prefer snapshotStartTimestamp over snapshotFinishTimestamp in case of a very long-running snapshot
+            // that started a long time ago
+            SnapshotInvocationRecord lastSuccess = policy.getLastSuccess();
+            return lastSuccess.getSnapshotStartTimestamp() != null
+                ? lastSuccess.getSnapshotStartTimestamp()
+                : lastSuccess.getSnapshotFinishTimestamp();
+        }
+        // SX TODO: handle first snapshot (i.e. no prior success of failure), maybe record the policy first trigger timestamp
```

Comment:
I think you can change this to just a regular TODO instead of one specific to you :)

Reply:
Don't forget about this one :)